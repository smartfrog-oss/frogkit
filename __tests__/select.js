import { mount } from 'vue-test-utils'
import Select from '@/components/select'

const options = [
  { label: 'Option 1', value:'Value 1'},
  { label: 'Option 2', value:'Value 2'},
  { label: 'Option 3', value:'Value 3'}
]

const placeholder = 'jest'

describe('Select component', () => {

  it('should render component and match snapshot', () => {
    const wrapper = mount(Select)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render disabled select', () => {
    const wrapper = mount(Select, { propsData: { disabled: true, options, placeholder } })
    expect(wrapper.html()).toMatchSnapshot()
  })

  describe('Select component | required', () => {
    it('should render with error', () => {
      const wrapper = mount(Select, { propsData : { value:'', required: true, options, placeholder } })
      wrapper.findAll('option').at(0).trigger('change')
      expect(wrapper.hasClass('fk-select--error')).toBe(true)
    })

    it('should render without error', () => {
      const wrapper = mount(Select, { propsData : { value:'tada', required: true, options, placeholder } })
      wrapper.findAll('option').at(1).trigger('change')
      expect(wrapper.hasClass('fk-select--error')).toBe(false)
    })
  })

})
