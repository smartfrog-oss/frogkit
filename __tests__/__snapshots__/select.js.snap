// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render component and match snapshot 1`] = `
<select class="fk-select">
  <!----> </select>
`;

exports[`Select component should render disabled select 1`] = `
<select class="fk-select">
  <option value="">jest</option>
  <option value="Value 1">
    Option 1
  </option>
  <option value="Value 2">
    Option 2
  </option>
  <option value="Value 3">
    Option 3
  </option>
</select>
`;
